### PR TITLE
Add user actions tracking

### PR DESCRIPTION
**Pull request by eliac123**
### What does this PR do?

A brief description of the change being made with this pull request.
Add user actions tracking

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [ ] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

File: packages/core/android/src/test/kotlin/com/datadog/reactnative/DdSdkTest.kt [hunk 1]
```diff
@@ -1039,6 +1039,70 @@ internal class DdSdkTest {
             }
     }
 
+    // TODO: Fix the next 2 tests to match with interactionTracking
+    @Test
+    fun `𝕄 initialize native SDK 𝕎 initialize() {no user action tracking}`(
```
Comment (xgouchet) [inline packages/core/android/src/test/kotlin/com/datadog/reactnative/DdSdkTest.kt:1044]: You're missing a test IMHO. You tested what should happen when `DD_NATIVE_INTERACTION_TRACKING` is set to `true` and `false`, but you should add a test for when it's set to `null` as well.

File: packages/core/android/src/test/kotlin/com/datadog/tools/unit/GenericAssert.kt [hunk 2]
```diff
@@ -49,6 +49,12 @@ class GenericAssert(actual: Any) :
         return this
     }
 
+    fun hasFieldClassEqualTo(name: String, expected: String): GenericAssert {
```
Comment (xgouchet) [inline packages/core/android/src/test/kotlin/com/datadog/tools/unit/GenericAssert.kt:52]: You should rename this function as `hasFieldWithClass`, and maybe have the param renamed to `expectedClassName` for clarity.

**Review by xgouchet — COMMENTED** [review]
Overall nice job, I left a couple of comments

File: packages/core/android/src/test/kotlin/com/datadog/tools/unit/GenericAssert.kt (hunk 2 above)
Comment (eliac123) [reply packages/core/android/src/test/kotlin/com/datadog/tools/unit/GenericAssert.kt:52]: Deal thanks, will change

File: packages/core/android/src/test/kotlin/com/datadog/tools/unit/GenericAssert.kt [hunk 3]
```diff
@@ -43,9 +43,15 @@ class GenericAssert(actual: Any) :
...
 
-    fun <F> hasFieldEqualTo(name: String, expected: F): GenericAssert {
+    fun <F> hasFieldEqualTo(name: String, expectedClassName: F): GenericAssert {
         val field: Any? = actual.getFieldValue(name)
-        assertThat(field).isEqualTo(expected)
+        assertThat(field).isEqualTo(expectedClassName)
+        return this
+    }
+
+    fun hasFieldWithClass(name: String, expected: String): GenericAssert {
```
Comment (louiszawadzki) [inline packages/core/android/src/test/kotlin/com/datadog/tools/unit/GenericAssert.kt:52]: change `expectedClassName` here

**Review by louiszawadzki — APPROVED** [review]
